elyra/pipeline/processor_local.py

```diff
@@ -0,0 +1,107 @@
+#
+# Copyright 2018-2020 IBM Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import os
+import papermill
+import tempfile
+import time
+
+from datetime import datetime
+from elyra.pipeline import PipelineProcessor, PipelineProcessorResponse
+from jupyter_client.kernelspec import NoSuchKernel
+
+
+class LocalPipelineProcessor(PipelineProcessor):
+    """
+    Local pipeline processor runs a pipeline locally. The scope of this runtime is
+    simply to automate the execution of multiple notebooks from a pipeline as if they
+    were being executed manually. Any additional support is out of the scope of this
+    processor. If a notebook don't run from your regular `run all` it will fails on the
+    same way when processed by this processor. Also, any relationship or specific capabilities
+    associated with a particular runtime is not supported by local mode and any additional properties
+    other then the specific file to be executed is ignored by the local processor.
+
+    Note: Execution happens in-place and a lodger of runs are available at $TMPFILE/elyra/pipeline-name-<timestamp>
+    """
+    _type = 'local'
+
+    def __init__(self, root_dir=None):
+        if root_dir:
+            self._root_dir = root_dir
+
+    @property
+    def type(self):
+        return self._type
+
+    def process(self, pipeline):
+        timestamp = datetime.now().strftime('%m%d%H%M%S')
+        pipeline_name = f'{pipeline.name}-{timestamp}'
+
+        pipeline_work_dir = self._get_pipeline_work_dir(pipeline_name)
+        self.log.info(f'Processing Pipeline : {pipeline.name} at {pipeline_work_dir}')
+
+        # TODO: consider ordering
+        for operation in pipeline.operations.values():
+            notebook = self.get_absolute_path(operation.filename)
+            assert os.path.isfile(notebook), "Invalid '{}'".format(notebook)
```

Comment:
Do we want an `AssertionError` here, or `FileNotFoundError`?  A message like "Invalid 'xyz.ipynb'" implies an invalid format of sorts, when the file is just not present.